File: app/src/main/java/com/UIN/Tool/plugin/PluginContext.java
package com.UIN.Tool.plugin;

import android.content.Context;
import android.content.ContextWrapper;
import android.content.res.AssetManager;
import android.content.res.Resources;
import android.view.LayoutInflater;

import com.UIN.Tool.utils.LogUtils;

import java.io.File;
import java.lang.reflect.Method;

public class PluginContext extends ContextWrapper {

    private final String pluginDir;
    private AssetManager pluginAssetManager;
    private Resources pluginResources;
    private LayoutInflater pluginInflater;

    public PluginContext(Context base, String pluginDir) {
        super(base);
        this.pluginDir = pluginDir;
        initPluginResources();
    }

    private void initPluginResources() {
        try {
            AssetManager assetManager = AssetManager.class.newInstance();
            Method addAssetPath = AssetManager.class.getMethod("addAssetPath", String.class);
            
            addAssetPath.invoke(assetManager, pluginDir);
            
            File resDir = new File(pluginDir, "res");
            if (resDir.exists()) {
                addAssetPath.invoke(assetManager, resDir.getAbsolutePath());
            }
            
            this.pluginAssetManager = assetManager;
            
            Resources superRes = super.getResources();
            this.pluginResources = new Resources(assetManager, superRes.getDisplayMetrics(), superRes.getConfiguration());
            // 修复：使用 getBaseContext() 而不是 base
            this.pluginInflater = LayoutInflater.from(getBaseContext());
            
            LogUtils.success("PluginContext", "插件资源加载成功: " + pluginDir);
        } catch (Exception e) {
            LogUtils.w("PluginContext", "资源加载失败: " + e.getMessage());
            this.pluginAssetManager = super.getAssets();
            this.pluginResources = super.getResources();
            // 修复：使用 getBaseContext()
            this.pluginInflater = LayoutInflater.from(getBaseContext());
        }
    }

    public String getPluginDir() { 
        return pluginDir; 
    }
    
    public String getPluginFilePath(String relativePath) { 
        return pluginDir + "/" + relativePath; 
    }
    
    public boolean hasPluginFile(String relativePath) { 
        return new File(pluginDir, relativePath).exists(); 
    }

    @Override
    public AssetManager getAssets() { 
        return pluginAssetManager != null ? pluginAssetManager : super.getAssets(); 
    }

    @Override
    public Resources getResources() { 
        return pluginResources != null ? pluginResources : super.getResources(); 
    }
    
    @Override
    public Object getSystemService(String name) {
        if (Context.LAYOUT_INFLATER_SERVICE.equals(name)) {
            return pluginInflater;
        }
        return super.getSystemService(name);
    }
}

File: app/src/main/java/com/UIN/Tool/plugin/PluginContext.kt
// ✅ 包名与 Java 版本完全一致：com.UIN.Tool.plugin
package com.UIN.Tool.plugin

import android.content.Context
import android.content.ContextWrapper
import android.content.res.AssetManager
import android.content.res.Resources
import android.view.LayoutInflater
import java.io.File

/**
 * 插件上下文 - 完全复用 Java 版本
 * Java 版本: PluginContext.java
 * 包名: com.UIN.Tool.plugin
 */
class PluginContext(
    base: Context,
    private val pluginDir: String
) : ContextWrapper(base) {

    private var pluginAssetManager: AssetManager? = null
    private var pluginResources: Resources? = null
    private var pluginInflater: LayoutInflater? = null

    init {
        initPluginResources()
    }

    private fun initPluginResources() {
        try {
            val assetManager = AssetManager::class.java.newInstance()
            val addAssetPath = AssetManager::class.java.getMethod("addAssetPath", String::class.java)

            // 添加插件目录
            addAssetPath.invoke(assetManager, pluginDir)

            // 添加资源目录
            val resDir = File(pluginDir, "res")
            if (resDir.exists()) {
                addAssetPath.invoke(assetManager, resDir.absolutePath)
            }

            this.pluginAssetManager = assetManager

            val superRes = super.getResources()
            this.pluginResources = Resources(assetManager, superRes.displayMetrics, superRes.configuration)
            this.pluginInflater = LayoutInflater.from(baseContext)

        } catch (e: Exception) {
            // 回退到宿主资源
            this.pluginAssetManager = super.getAssets()
            this.pluginResources = super.getResources()
            this.pluginInflater = LayoutInflater.from(baseContext)
        }
    }

    fun getPluginDir(): String = pluginDir

    fun getPluginFilePath(relativePath: String): String = "$pluginDir/$relativePath"

    fun hasPluginFile(relativePath: String): Boolean = File(pluginDir, relativePath).exists()

    override fun getAssets(): AssetManager {
        return pluginAssetManager ?: super.getAssets()
    }

    override fun getResources(): Resources {
        return pluginResources ?: super.getResources()
    }

    override fun getSystemService(name: String): Any? {
        if (Context.LAYOUT_INFLATER_SERVICE == name) {
            return pluginInflater ?: super.getSystemService(name)
        }
        return super.getSystemService(name)
    }
}